ui.apps/src/main/content/jcr_root/apps/venia/components/structure/navigation/v1/navigation/clientlibs/js/navigation.js

```diff
@@ -0,0 +1,110 @@
+/*******************************************************************************
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+(function () {
+    "use strict";
+    const selectors = {
+        navigationTrigger: '.header__primaryActions .navTrigger__root',
+        navigationRoot: 'aside.navigation__root',
+        shadowTreeRoot: '.categoryTree__root--shadow',
+        categoryTreeRoot: '.categoryTree__root',
+        activeNavigation: '.categoryTree__root .categoryTree__tree',
+        shadowNavigations: '.categoryTree__root--shadow .categoryTree__tree',
+        backNavigationButton: '.navigation__header .trigger__root--back',
+        backNavigationEmpty: '.navigation__header .trigger__root--back--empty',
+        closeNavigationButton: '.navigation__header .trigger__root--close',
+        downNavigationButton: '.categoryLeaf__root button'
+    };
+
+    const CSS_NAVIGATION_OPEN = 'navigation__root_open';
+
+    let backNavigationButton;
+    let backNavigationEmpty;
+    let categoryTreeRoot;
+    let shadowTreeRoot;
+    let navigationPanel;
+
+    function setNavigationPanelVisible(visible) {
```

Comment:
Having `show()`/`hide()` functions to handle visibility makes for a clearer code.